src/main/java/com/wiley/autotest/actions/RepeatableAction.java

```diff
@@ -2,49 +2,87 @@
 
 import com.wiley.autotest.utils.TestUtils;
 
+import java.util.function.Supplier;
+
 /**
  * Repeats action limited number of times until a boolean condition is true
  * waits for a timeout between attempts to perform an action
  * <p>
  * If the condition does not become true after all attempts - AssertionError is thrown
  */
 public class RepeatableAction {
-    private final Actions action;
-    private final Conditions condition;
+    private Actions action;
+    private Conditions condition;
+    private Supplier<Boolean> supplier;
     private final int numberOfAttempts;
     private final int millisecondsBetweenAttempts;
     private int attemptCounter = 0;
+    private String errorMessage;
 
     /**
      * By default will try 5 types and sleep 3 seconds after each attempt
      *
-     * @param action    - any function you want yo be performed
+     * @param action    - any function you want to be performed
      * @param condition - any boolean function to check after action
      */
     public RepeatableAction(Actions action, Conditions condition) {
         this(action, condition, 5, 3000);
     }
 
+    /**
+     * By default will try 5 types and sleep 3 seconds after each attempt
+     *
+     * @param action - any boolean function you want to be performed
+     */
+    public RepeatableAction(Supplier<Boolean> action) {
```

Comment:
this var name is "action" but in the next constructor you call it a "supplier". Use one name... what about "action"?

Reply:
Renamed to supplier because of action field here